### PR TITLE
Suggestions

### PR DESCRIPTION
**Pull request by david-littlefarmer**
Here is list of suggestions:

1. Makefile
2. Linter
3. `common.go`
    - fixed parsing of rpc request to work with paths including network ex: `http://localhost:4422/polygon/rpc/Marketplace/ListCurrencies`
    - simplified check for method name in `func (c Config[T]) Get(r *rcpRequest) (v T, ok bool)`
4. `context.go`
    - added comments
    - changed function names, so they are not exported, since we don't need it. Or do we need them exported?
5. `middleware.go`
    - returned `admin` session to switch, is that right? 
6. `proto`
    - bumped versions to latest webrpc v0.21.0
    - generating only clients, we don't need server
7. github action go version from `go.mod` file
8. Test cases for invalid requests, expired jwt etc.

### CODE REVIEW

File: common.go [hunk 1]
```diff
@@ -29,70 +29,69 @@ type UserStore interface {
...
 
 // Config is a generic map of services/methods to a config value.
+// map[service]map[method]T
 type Config[T any] map[string]map[string]T
 
-// Get returns the config value for the given request.
-func (c Config[T]) Get(r *rcpRequest) (v T, ok bool) {
-	if c == nil || r.Package != "rpc" {
+// returns the config value for the given request.
+func (c Config[T]) get(r *rcpRequest) (v T, ok bool) {
```
Comment (klaidliadon) [inline common.go:36]: Can you keep this public?
I use it here: https://github.com/0xsequence/quotacontrol/pull/53/files#diff-7bf08ce0a790451f4677b47d7e17aafcc1825aa92312c83775de6c22e7b5a102R10-R23

File: common.go [hunk 2]
```diff
@@ -29,70 +29,69 @@ type UserStore interface {
...
+	if c == nil {
 		return v, false
 	}
-	serviceCfg, ok := c[r.Service]
-	if !ok {
-		return v, false
-	}
-	methodCfg, ok := serviceCfg[r.Method]
+
+	methodCfg, ok := c[r.serviceName][r.methodName]
```
Comment (klaidliadon) [inline common.go:41]: if the name of the service is invalid, this panics

File: common_test.go [hunk 3]
```diff
@@ -7,19 +7,21 @@ import (
...
 
-	"github.com/0xsequence/authcontrol/proto"
 	"github.com/go-chi/jwtauth/v5"
 	"github.com/stretchr/testify/require"
+
+	"github.com/0xsequence/authcontrol/proto"
 )
 
-func mustJWT(t *testing.T, auth *jwtauth.JWTAuth, claims map[string]any) string {
+func mustJWT(t *testing.T, auth *jwtauth.JWTAuth, claims map[string]any) *string {
```
Comment (klaidliadon) [inline common_test.go:16]: There is no need for a pointer here, empty string is enough

File: context.go [hunk 4]
```diff
@@ -23,8 +23,12 @@ var (
 	ctxKeyProjectID   = &contextKey{"ProjectID"}
 )
 
-// WithSessionType adds the access key to the context.
-func WithSessionType(ctx context.Context, accessType proto.SessionType) context.Context {
```
Comment (klaidliadon) [inline context.go:27]: We can't make this private, we need this public for a special case.

File: context.go [hunk 5]
```diff
@@ -59,8 +71,12 @@ func GetUser[T any](ctx context.Context) (T, bool) {
...
 }
 
-// WithService adds the service to the context.
-func WithService(ctx context.Context, service string) context.Context {
+//
+// Service
+//
+
+// withService adds the service to the context.
+func withService(ctx context.Context, service string) context.Context {
```
Comment (klaidliadon) [inline context.go:79]: Same here, please keep this public for now

**Review by klaidliadon — COMMENTED** [review]
Great addition overall, a few notes

File: common.go (hunk 2 above)
Comment (david-littlefarmer) [reply common.go:41]: This doesn't panic, we have same pattern in marketplace: https://goplay.tools/snippet/MkIt6rVZGft
I also added this check with invalid service name to tests and it is passing.

File: common_test.go (hunk 3 above)
Comment (david-littlefarmer) [reply common_test.go:16]: This is for testing purposes, i need o pointer so i am able to send just "Bearer " in testing cases.

File: middleware.go [hunk 6]
```diff
@@ -82,34 +88,37 @@ func Session(auth *jwtauth.JWTAuth, o *Options) func(next http.Handler) http.Han
...
 						sessionType = proto.SessionType_Admin
-						break
 					}
 
 					if projectClaim > 0 {
 						projectID := uint64(projectClaim)
 						ctx = withProjectID(ctx, projectID)
 						sessionType = proto.SessionType_Project
 					}
+				case adminClaim:
```
Comment (klaidliadon) [inline middleware.go:111]: If you add this we accept a JWT with only admin claim, which is not correct.
I expect admin+address which is handled in the `accountClaim` block.

**Comment by david-littlefarmer** [comment]
All comments addressed.
We shouldn't export setters for `WithSessionType` and `WithService`, these should be set only by this package, and not have the possibility from anywhere else. We need to tackle this in future.

**Review by klaidliadon — APPROVED** [review]

**Pull request merged**